buildtools/maven/maven.go

```diff
@@ -59,53 +72,136 @@ func (m *Maven) Compile(dir string) error {
 	return err
 }
 
-func Modules(dir string) ([]string, error) {
+// ResolveManifestFromBuildTarget tries to determine what buildTarget is supposed to be and then reads the POM
+// manifest file pointed to by buildTarget if it is a path to such a file or module.
+func (m *Maven) ResolveManifestFromBuildTarget(buildTarget string) (*Manifest, error) {
+	var pomFile string
+	if !strings.HasSuffix(buildTarget, ".xml") {
+		pomFile = filepath.Join(buildTarget, "pom.xml")
+	} else {
+		pomFile = buildTarget
+	}
+	if ok, _ := files.Exists(pomFile); !ok {
+		if strings.Count(buildTarget, ":") == 1 {
+			// This is likely a module ID.
+			return nil, fmt.Errorf("build target %q appears to be a module identifier", buildTarget)
+		}
+		return nil, fmt.Errorf("module manfist file %q does not exist", buildTarget)
+	}
+
 	var pom Manifest
-	err := files.ReadXML(&pom, filepath.Join(dir, "pom.xml"))
-	if err != nil {
-		return nil, errors.Wrap(err, "could not read POM file")
+	if err := files.ReadXML(&pom, pomFile); err != nil {
+		return nil, err
+	}
+	if m.manifestCache == nil {
+		m.manifestCache = make(map[string]*Manifest)
+	}
+	m.manifestCache[buildTarget] = &pom
```

Comment:
I'm not convinced the manifest cache is the best way to go about this, it doesn't appear to be working on my end because the manifest cache is never accessed before we return the dependencies. We should discuss whether or not this is the way to proceed, I think this might be optimizing a little early.

Reply:
please try this out with the latest commits of this branch.

As for whether there should even be a `manifestCache`, the thing is that sometimes we'd need to access the data in the XML file at two separate times though closely in time. These manifest files can be very large, and re-parsing the XML can be expensive. What do you think? If not worth it let me know.

Reply:
Ah, so I see that this does update `graph.Deps.Direct`, we also need to update `graph.Deps.Transitive` in order to make sure that that core picks up the correct dependency graph.